pkg/apis/pipeline/v1beta1/artifact_types.go

```diff
@@ -0,0 +1,21 @@
+package v1beta1
+
+// Algorithm Standard cryptographic hash algorithm
+type Algorithm string
+
+// Artifact represents an artifact within a system, potentially containing multiple values
+// associated with it.
+type Artifact struct {
+	Name   string          `json:"name,omitempty"`   // The artifact's identifying category name
+	Values []ArtifactValue `json:"values,omitempty"` // A collection of values related to the artifact
+}
+
+// ArtifactValue represents a specific value or data element within an Artifact.
+type ArtifactValue struct {
```

Comment:
That's right. StepAction was a new CRD. This is not.

Regarding type alias between `v1` and `v1beta1`, I completely agree with you. We don't need duplication in `v1beta1`.

Reply:
Hi @vdemeester , instead of aliasing, can we use v1 reference directly on v1beta1 struct, like this? 
```golang
type StepState struct {
	corev1.ContainerState `json:",inline"`
	Name                  string              `json:"name,omitempty"`
	ContainerName         string                `json:"container,omitempty"`
	ImageID               string                `json:"imageID,omitempty"`
	Results               []TaskRunStepResult   `json:"results,omitempty"`
	Inputs               []v1.TaskRunStepArtifact `json:"inputs,omitempty"`
	Outputs               []v1.TaskRunStepArtifact `json:"outputs,omitempty"`
}
``

Reply:
```
Side note: why we do not use type aliases between v1 and v1beta1 ?
```

after thinking about this, using aliases can lead some problems, 

If we only use one high level parent struct alias,  
``type Parent v1.Parent``, this v1beta1.Parent will contain v1 fields, this can make type conversion from v1beta1 to v1 very challenging. For example, a struct FOO on v1beta1 contains v1beta1 Parent, when we convert v1beta1 FOO to v1 FOO, we need to do a deepCopy of the Parent field, if this fields has just a few primitive fields, that's probably fine. However, a struct can grow very complicated, and if that is the case we cannot implement type conversion methods(like convertTo) on v1 struct from v1beta1 package, without a generic method, doing deepCopy on a nested struct is error-prune.

Reply:
Ah right, I remember something like this indeed 🙃